synapse/datamodel.py

```diff
@@ -250,6 +251,18 @@ def getStorNode(self, form):
 
         return (buid, {'props': pnorms, 'ndef': ndef})
 
+    def getAlts(self):
+        '''
+        Return a list of Prop instances that are considered
+        alternative locations for our property value, including
+        self.
+        '''
+        if self.alts is None:
+            self.alts = [self]
```

Comment:
does this need a weakref?

Reply:
why would it?

Reply:
I thought maybe because of the circular reference but if you don't think so then nm.